cdi-core/src/main/java/com/linkedin/cdi/source/S3SourceV2.java

```diff
@@ -130,4 +135,33 @@ private String getEndpointFromHost(String host) {
     segments.remove(0);
     return Joiner.on(".").join(segments);
   }
+
+  /**
+   *
+   * @param parameters JsonObject containing ms.source.s3.parameters
+   * @param host hostname with bucket name in the beginning
+   * @return the bucket name
+   */
+  @VisibleForTesting
+  protected String getBucketName(JsonObject parameters, String host) {
+    if (parameters.has(KEY_BUCKET_NAME)) {
+      return parameters.get(KEY_BUCKET_NAME).getAsString();
+    }
+    return host.split("\\.")[0];
+  }
+
+  /**
+   *
+   * @param parameters JsonObject containing ms.source.s3.parameters
+   * @param host hostname with bucket name in the beginning
+   * @return the endpoint name without the bucket name
```

Comment:
nit: in the comment maybe add some details to explain its difference from getEndpointFromHost. That will help with debugging in the future

Reply:
Sure, I will add